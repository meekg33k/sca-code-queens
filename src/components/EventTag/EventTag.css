.largeTag {
  background-color: #1abc9c;
  display: flex;
  justify-items: center;
  justify-content: center;
  padding: 12px;
  width: 4em;
  border-radius: 10px;
}
.largeTag > p {
  position: relative;
  left: 2px;
  bottom: -2px;
}
.smallTag{
  width: 4.2em;
  padding: 3px;
  background-color: #1abc9c;
  display: flex;
  border-radius: 5px;
  justify-items: center;
  justify-content: center;
}
.opac{
  opacity: 0.5;
}
.smallTag > p{
  font-size: 0.8em !important;

}
.smallTag > img{
  width: 1em;
}