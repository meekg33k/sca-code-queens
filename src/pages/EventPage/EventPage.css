body {
    overflow-x: hidden;
}

.djDetails {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    margin-top: 15px;
}

.photoCover {
    flex-basis: 60%;
}

.eventDetail {
    flex-basis: 35%;
}

@media (max-width: 740px) {
    .eventDetail {
      order: 1;
      flex-basis: 100%;
      margin-left: 45px;
      margin-top: 15px;
    }
    .photoCover {
      order: 2;
      flex-basis: 100%;
      margin-right: 45px;
    }
  }