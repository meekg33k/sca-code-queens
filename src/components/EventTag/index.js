import React from "react";
import PropTypes from "prop-types";
import Icon from "../Icon";
import Text from "../Text/index";
import "./EventTag.css";

const EventTag = ({ icon, partyType, attendeeNumber, size, opacity}) => {
  
  let attendee = attendeeNumber ?  attendeeNumber.toString() : false
  let opac = opacity === true ? "opac" : false
  let style = size==="smallTag" ? "smallTag" : "largeTag"

  return (
      <div className= {`${style} ${opac}`} >
          <Icon src={icon} alt={`${partyType} icon`} />
          <Text
          text={partyType || attendee}
          size="default"
          color="white"
          />      
        </div>
  )
};

EventTag.defaultProps = {
  icon: "",
  // // partyType: "partytype or attendee",
  // attendeeNumber: 0
};

EventTag.propTypes = {
  icon: PropTypes.string,
  attendeeNumber: PropTypes.number,
  partyType: PropTypes.string ,
  size: PropTypes.string ,
  opac: PropTypes.string ,
};

export default EventTag;
