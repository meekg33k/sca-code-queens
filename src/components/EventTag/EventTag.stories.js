import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import EventTag from ".";
import { peopleIcon } from "../../pages/EventPage/index";


storiesOf("EventTag", module)
  .addDecorator(
    withInfo({
      text: "This is a component renders an event Tag showing the event type or attendee number. The options for size are largeTag or smallTag"
    })
  )
  .addDecorator(withKnobs())
  .add("EventTag", () => {
    return <EventTag 
    icon={peopleIcon}
    attendeeNumber={100}
    size={"largeTag"}
    opacity={false}/>;
  })
  .add("your Event tag", () => {
    const icon = text("path", peopleIcon);
    const partyType = text("partType", 'people');
    const size = text("size", "smallTag");
    const opacity = boolean('opacity', true)

    return  <EventTag 
    icon={icon}
    partyType={partyType}
    size={size}
    opacity={opacity}/> ;
  });
