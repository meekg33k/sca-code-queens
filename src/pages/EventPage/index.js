import React from "react";
import Footer from "../../components/Footer";
import Navigation from "../../components/Navigation";
import EventDetail from "../../components/EventDetail";
import CoverPhoto from "../../components/CoverPhoto";
import EventTag from "../../components/EventTag";
import Location from "../../components/Location";
import "./EventPage.css";

export const peopleIcon = "./assets/people.svg"

export const EventPage = () => {
  return (
    <div className="event-page">
      
      <Navigation />
      
      <EventTag 
      icon={peopleIcon}
      attendeeNumber={100}
      size={"largeTag"}
      />
      <EventTag 
      icon={peopleIcon}
      attendeeNumber={100}
      size={"smallTag"}
      opacity={true}/>
      <EventTag 
      icon={peopleIcon}
      partyType={"wedding"}
      size={"smallTag"}
      opacity={false}/>
      <Location 
        name= "DJ Spinal Take Over" 
        location="Party Club Estate, Alalaba, Lagos" 
      />
      <div className="djDetails">
        <div className="photoCover">
            <CoverPhoto name="DJ Spinal" coverimg="./assets/smiley.jpg" />
        </div>
        <div className="eventDetail">
            <EventDetail 
            about={"DJ Spinal shutting Down popular club in Yaba"}
            time={"8:00 PM - Dawn"}
            date={"28th January, 2020"}/>
        </div>
      </div>
      
      <Footer />
      
    </div>
  );
};

export default EventPage;
