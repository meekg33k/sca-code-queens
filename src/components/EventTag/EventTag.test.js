import React from "react";
import { mount } from "enzyme";
import { peopleIcon } from "../../pages/EventPage/index";

import EventTag from "./index";
const setUp = () => {
    return mount( <EventTag 
        icon={peopleIcon}
        attendeeNumber={100}
        size={"largeTag"}
        opacity={true}/>);
  };

describe("EventTag", () => {
    let Tag
  beforeEach(() => {
     Tag = setUp();
  });
  afterEach(() => {
    Tag.unmount();
  });

    it("should match snap shot", () => {
      
      expect(Tag).toMatchSnapshot();
    });
    it("contains icon ", () => {
        const icon = Tag.find("img");
    
        expect(icon.length).toBe(1);
    });
    it("contains text ", () => {
        const text = Tag.find("p");
    
        expect(text.length).toBe(1);
        expect(text.text()).toEqual("100");
      });
});
