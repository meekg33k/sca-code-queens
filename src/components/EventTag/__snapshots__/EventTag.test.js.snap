// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTag should match snap shot 1`] = `ReactWrapper {}`;

exports[`EventTag snapshot should match snap shot 1`] = `ReactWrapper {}`;
